Source/GameView/Models/GameModel.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  GameModel.swift
+//  Feisty
+//
+//  Created by Tiewhan Smith on 2020/03/02.
+//  Copyright © 2020 DVT. All rights reserved.
+//
+
+import Foundation
+
+///The GameModel class retrieves data when asked by the View Model
+public class GameModel {
+  
+  private let repo: GameRepository
+  public var observerID: String = "GameModelObserver"
+  
+  public init(_ repo: GameRepository) {
+    self.repo = repo
+    
+    repo.subscribeToGameRepoGamesLoaded(subscriber: self, subscriberID: observerID)
+    
+  }
+  
+  public var gameList: [Game] = []
+  
+  private var observers: [String: GameModelObserver] = [:]
+  
+  private func notifyAllObservers() {
+    observers.forEach({ (observer) in
+      observer.value.gamesFinishedLoading()
+    })
+  }
+  
+  private func loadGameDetails(of gameList: [Game]) {
+    repo.getGameDetails(of: gameList, with: ServiceCaller())
+  }
+  
+}
+
+extension GameModel: GameModelProtocol {
+  
+  public func getPageCount() -> Int {
```

Comment:
Instead of having a function when there is little logic aside from a return, we can instead have:
```
public var pageCount: Int {
  return gameList.count
}
```

Just my opinion here but it's more intuitive to have a variable as setters and getters are typically more for Java